lib/Runtime/Library/JavascriptNumber.cpp

```diff
@@ -235,7 +235,35 @@ namespace Js
         }
         else if(TryGetInt32Value(y, &intY))
         {
-            return ::pow(x, intY);
+            // For exponent in [-8, 8], aggregate the product according to binary representation
+            // of exponent. This acceleration may lead to significant deviation for larger exponent
+            if (intY >= -8 && intY <= 8)
+            {
+                uint32 uexp;
+                if (intY >= 0)
+                {
+                    uexp = static_cast<uint32>(intY);
+                }
+                else
+                {
+                    uexp = static_cast<uint32>(-intY);
+                }
```

Comment:
I think using the ternary operator here would make this shorter and easier to read, i.e. `uint32 uexp = static_cast<uint32>(intY >= 0 ? intY : -intY)`